app/workers/bgs/submit_form674_job.rb

```diff
@@ -11,23 +11,23 @@ class Invalid674Claim < StandardError; end
 
     sidekiq_options retry: false
 
-    def perform(user_uuid, saved_claim_id, vet_info)
-      user = User.find(user_uuid)
-      Rails.logger.info('BGS::SubmitForm674Job running!', { user_uuid:, saved_claim_id:, icn: user&.icn })
+    def perform(user_uuid, icn, saved_claim_id, vet_info)
+      Rails.logger.info('BGS::SubmitForm674Job running!', { saved_claim_id:, icn: })
       in_progress_form = InProgressForm.find_by(form_id: FORM_ID, user_uuid:)
       in_progress_copy = in_progress_form_copy(in_progress_form)
       claim_data = valid_claim_data(saved_claim_id, vet_info)
       normalize_names_and_addresses!(claim_data)
+      temp_user = generate_temp_user(vet_info['veteran_information'])
```

Comment:
(Non blocking) I might reconsider calling this a "temp user", although I'm not really sure what a better name would be. Maybe user metadata? User hash? since this is just a data structure you are storing user info in as opposed to an actual user record.

Reply:
changed the name to user_struct